df_chat/migrations/0003_alter_room_users.py

```diff
@@ -0,0 +1,32 @@
+# Generated by Django 4.2.3 on 2023-07-11 04:18
+
+from django.conf import settings
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ("df_chat", "0002_remove_roomuser_is_online_userchat"),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                # Old table name from checking with sqlmigrate, new table
+                # name from AuthorBook._meta.db_table.
+                migrations.RunSQL(sql="ALTER TABLE df_chat_room_users RENAME TO df_chat_removable_room_users",
```

Comment:
@Ambitiont109  I don't get the logic of this state operation. Could you please explain it? 

I thought we can just add through parameter and have a default generated migration.

Reply:
When you run default generated migration for newly added ‘through’ field to your M2M relation, you got the following error.
ValueError: Cannot alter field xxx into yyy they are not compatible types (you cannot alter to or from M2M fields, or add or remove through= on M2M fields)
To resolve it, we have to make the custom migration file.

Reply:
@Ambitiont109 the package is not stable yet, you can just have two migrations: for deleting field and for creating a new one

Reply:
@eugapx  which field has to be deleted?
There is nothing to be deleted. It is only necessary to change the state of the Model since we only added the through attribute to the `users` field in the `room` Model.